HZN-1182: Add more configuration possibilities to elastic jest client factory

mvrueden opened the pull request:
* JIRA: https://issues.opennms.org/browse/HZN-1182

j-white commented:
Looks good - just need to fix those tests.

j-white commented on opennms-base-assembly/src/main/filtered/etc/elastic-flow-persistence-credentials.xml:
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Can we rename this to something more generic like  `elastic-credentials.xml`? This would allow us to reuse in other contexts, since there isn't anything specific to flows here.

mvrueden replied:
Yes, sure, I just have to place it somewhere else then.

mvrueden commented:
I changed the name to `elastic-credentials.xml` and also moved the credentials files to the generic `jest.client` module. This looks now much simpler. :)

mvrueden commented:
I rebased against `features/drift` and made the tests pass.
